Package-Level Makefiles for Testing

akutz opened the pull request:
This patch introduces package-level Makefiles for driving tests for the libStorage framework as well as the storage drivers.

codenrhoden reviewed (commented):
Is there any way to not duplicate all the package level Makefiles? a symlink to a master version or something? I've never committed a symlink to git, though I know it's possible, but I'm not sure of the ramifications.

I do like all the latest changes. will make the project easier to work on by all, and more approachable.

akutz commented:
Hi @codenrhoden,

A symlink is possible and so is placing a file at the top level called `driver-tests.mk` or `framework-tests.mk` and then in the individual directories including a `Makefile` that has a single line:

```Makefile
include ../../../driver-tests.mk
```

Or something like that.

The reason I didn't opt for referencing a file at the root of the project is for portability. We plan to break the drivers out into other repos, and so I didn't want to link to anything at the root of this one. Does that make sense?

akutz committed:
Package-Level Makefiles for Testing

This patch introduces package-level Makefiles for driving tests for the
libStorage framework as well as the storage drivers.

codenrhoden reviewed (approved):
More simplifications! I love it.

I know that shows I don't understand go coverage all that well, but why isn't `TEST_COVERPKG` necessary anymore?
